src/Orleans.Runtime/Hosting/CoreHostingExtensions.cs

```diff
@@ -95,11 +110,62 @@ public static ISiloHostBuilder ConfigureSiloName(this ISiloHostBuilder builder,
         }
 
         /// <summary>
-        /// Configure silo to use Development membership
+        /// Configures a localhost silo for development and testing.
         /// </summary>
-        public static ISiloHostBuilder UseDevelopmentClustering(this ISiloHostBuilder builder, Action<DevelopmentMembershipOptions> configureOptions)
+        /// <param name="builder">The silo builder.</param>
+        /// <param name="siloPort">The silo port.</param>
+        /// <param name="gatewayPort">The gateway port.</param>
+        /// <param name="primarySiloEndpoint">
+        /// The endpoint of the primary silo, or <see langword="null"/> to use this silo as the primary.
+        /// </param>
+        /// <returns>The silo builder.</returns>
+        public static ISiloHostBuilder UseLocalhostClustering(
```

Comment:
Can we keep development extensions in a different namespace.  Maybe Orleans.Development?  This would hide them unless the user specifically opts into them for development purposes, and would prevent users from being confused by them when writing their production logic.

Reply:
Users wont find them if they're not in the right namespace. Is `UseLocalhostClustering` not obvious enough that it's not a scale-out solution?

Reply:
To be clear, I'm ok with the existing namespace, just suggesting we use a different one.

As far as users not finding them, that, imo, was the goal.  I think it's good to have developer only options less discoverable than the expected production behaviors.  Should one include the development namespace, they will be equally discoverable.

As far as the name being clear enough, I was suggesting this as a general pattern for all developer features, not just these calls.  For instance, extensions setting up the grain based membership service, the memory storage provider, the in-memory transaction log, or any other developer tooling would all be exposed under a development namespace, communicating clearly and in a consistent way what features we've provided for development purposes vs. production use.

This is a suggestion, entirely optional and not something we should block this PR for.

Reply:
Ok, got it.
I think we should do something to make sure that users explicitly opt-in to enabling features which we deem unsuitable for production use.

We could use `HostBuilderContext` (eg `Properties` or `HostingEnvironment`) to signal that the developer has enabled "dev/test only" features.